1. Print out your favorite color and write comment about what you are doing above the code.

In [70]:
# ask a user to enter what their favorite color is

color = input("Enter your favorite color: ")
print(f"My favorite color is {color}.")

Enter your favorite color: blue
My favorite color is blue.


In [72]:
print("My favorite color is blue.")

My favorite color is blue.


2. Print the sumation of numbers 1 through 10

In [69]:
print(sum(range(11)))



55


3. Print what 2 to the 8th power is.

In [1]:
print(2**8)

256


4. Print the remainder of 100 divided by 11. 

In [7]:
100 % 11

1

5. What is the result of adding strings: 20, 8, 77?

In [8]:
print('20' + '8' + '77')

20877


6. Print out 100 divided by 5

In [11]:
print("100 / 5")
print(100 / 5)

100 / 5
20.0


7. What is the difference (subtraction) between 10/4 and 10//4?  Print out the result

In [12]:
print((10 / 4) - (10 // 4))

0.5


8. Create a variable called height and store your height value there.
Print a sentence describing your height and use the variable in the sentence as well.

In [18]:
height = input("Enter your height(cm): ")
print(f"My height is (cm): {height}")

Enter your height(cm): 60
My height is (cm): 60


9. Print the same information as exercise 8 using a formatted string. 

In [ ]:
height = input("Enter your height(cm): ")
print(f"My height is (cm): {height}")

10. Use two different (single and double) quotes in a string to print out a sentence

In [73]:
print("Hello, she said. 'Nice to meet you', he replied")

Hello, she said. 'Nice to meet you', he replied


11. Print out Hello World! 7 times each on a new line using only one line of code.

In [19]:
print("Hello" * 7)

HelloHelloHelloHelloHelloHelloHello


12. Write a code to get two integers from a user. Then print out their summation, subtraction, multiplication, and division.

In [28]:
value1 = int(input("Enter your first number: "))
value2 = int(input("Enter your second number: "))
print(value1 + value2)
print(value1 - value2)
print(value1 * value2)
print(value1 / value2)

Enter your first number: 12
Enter your second number: 20
32
-8
240
0.6


13. You and your collegue have been tasked to write a code to get 5 numbers from a user and then print out their summation and average in a formatted string.  Your partner wrote codes below.  Now it is up to you to fix all errors.

In [35]:
prompt = (f"Enter a number: ")

numb1 = int(input("Enter a number: "))
numb2 = int(input("Enter a number: "))
numb3 = int(input("Enter a number: "))
numb4 = int(input("Enter a number: "))
numb5 = int(input("Enter a number: "))

total = numb1 + numb2 + numb3 + numb4 + numb5
avg = int(total) / 5

print(f"{numb1} + {numb2} + {numb3} + {numb4} + {numb5} = {total}")
print(f"average is {avg}")

Enter a number: 12
Enter a number: 20
Enter a number: 15
Enter a number: 9
Enter a number: 4
12 + 20 + 15 + 9 + 4 = 60
average is 12.0


14. You and your partner have been assigned to a project to write a program to calculate the amount of runoff rain on a roof from any given rainfall.  

You and your partner have figured out that to calculate the runoff from any given rainfall. 

You need to take the dimensions of the footprint of the roof and <b>convert them to inches</b> (ex - a 50' x 20' roof is 600" x 240"). The dimensions should be user submitted. 

Then, multiply the roof dimensions by the number of inches of rainfall. As an example, 600" x 240" x 1" = 144,000 cubic inches of water for an inch of rainfall. Finally, divide that result by 231 to get the number of gallons (because 1 gallon = 231 cubic inches; 144,000/231 = 623.38).

Your partner started coding before getting sick and it is up to you to finish the program.

In [75]:
print("\nRainfall Calculation ***\n")

width = int(input("What's roof's width in feet? "))
length = int(input("What's roof's length in feet? "))
rain = int(input("How much did it rain in inches? "))

roofd = int(width * 12)
area = int(roofd * length)
rainfall = int(area * rain)
gallons = int(rainfall / 231)

print(f"The amount of rain runoff is: {gallons} gallons")




Rainfall Calculation ***

What's roof's width in feet? 100
What's roof's length in feet? 80
How much did it rain in inches? 2
The amount of rain runoff is: 831 gallons


15. A program is required to get a customer’s name, a purchase amount and a discount rate (in %). The program must compute the discount amount, sales tax (6%) and the total amount due. Using one print statement, print the customer’s name, purchase amount, discount amount, sales tax and total amount due in friendly format.

In [80]:
cust_name = input("Enter your full name: ")
purchase = float(input("Enter purchase amount: "))
discount_rate = int(input("Enter discount %: "))
tax_rate = .06
tax = float((purchase - discount) * tax_rate)
discount = float(purchase * (discount_rate / 100))

total = purchase - discount + tax
print(f"Your discount is: ${discount:.2f}")
print(f"Your tax amount is: ${tax:.2f}")
print(f"{cust_name}, the total amount due is: ${total:.2f}")

Enter your full name: Tania Abbey
Enter purchase amount: 42.67
Enter discount %: 12
Your discount is: $5.12
Your tax amount is: $2.25
Tania Abbey, the total amount due is: $39.80


In [83]:
print(f"{cust_name}, your purchase amount is ${purchase:.2f},\nyour discount is ${discount:.2f},\nyour sales tax is ${tax:.2f},\nand the total due is ${total:.2f}.")

Tania Abbey, your purchase amount is $42.67,
your discount is $5.12,
your sales tax is $2.25,
and the total due is $39.80.
